libs/common/src/platform/services/user-key-init.service.ts

```diff
@@ -0,0 +1,57 @@
+import { EMPTY, Subscription, catchError, filter, from, switchMap } from "rxjs";
+
+import { AccountService } from "../../auth/abstractions/account.service";
+import { UserId } from "../../types/guid";
+import { CryptoService } from "../abstractions/crypto.service";
+import { LogService } from "../abstractions/log.service";
+import { KeySuffixOptions } from "../enums";
+
+// TODO: this is a half measure improvement which allows us to reduce some side effects today (cryptoService.getUserKey setting user key in memory if auto key exists)
+// but ideally, in the future, we would be able to put this logic into the cryptoService
+// after the vault timeout settings svc is transitioned to state provider so that
```

Comment:
please avoid these abbreviated words.

```suggestion
// after the vault timeout settings service is transitioned to state provider so that
```